calculator/src/test/java/com/market/banica/calculator/service/BackUpServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package com.market.banica.calculator.service;
+
+import com.market.banica.calculator.data.contract.ProductBase;
+import com.market.banica.calculator.enums.UnitOfMeasure;
+import com.market.banica.calculator.model.Product;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class BackUpServiceImplTest {
+
+    private final String DATABASE_BACKUP_URL = "src/test/resources/backUpRecipeBase.json";
+    public static final String PRODUCT_NAME = "crusts";
+
+    private File testFile;
+    private ConcurrentHashMap<String, Product> base;
+
+    @Mock
+    private ProductBase productBase;
+
+    private BackUpServiceImpl backUpService;
+
+    @BeforeEach
+    void setUp() {
+        backUpService = new BackUpServiceImpl(productBase);
+        ReflectionTestUtils.setField(backUpService, "databaseBackUpUrl", DATABASE_BACKUP_URL);
+        base = new ConcurrentHashMap<>();
+        base.put("crusts", createProduct());
+    }
+
+    @AfterEach
+    void tearDown() {
+        Paths.get(DATABASE_BACKUP_URL).toFile().delete();
+    }
+
+    @Test
+    void readBackUpShouldCreateNewFileIfBackUpFileDoesNotExists() {
+        //Arrange//Act
```

Comment:
if you don't have any data to be prepared, Arrange should not be mentioned at all